floating: Maximize animation

Drakulix opened the pull request:
And a bunch of fixes for geometry restoring in floating mode.

wash2 commented:
This seems mostly good, but for some reason I see a compositor panic when I maximize a window in tiling mode, then switch to window mode, unmaximizing the window, and then drag the window.

wash2 commented:
thread 'main' panicked at 'assertion failed: was_floating != was_tiled.is_some()': src/shell/mod.rs:2292

Drakulix commented:
> This seems mostly good, but for some reason I see a compositor panic when I maximize a window in tiling mode, then switch to window mode, unmaximizing the window, and then drag the window.

I assume "window mode" should have been "floating mode". I can't reproduce it with the last commit anymore, can you verify? Basically I think what has happened was switching to "floating mode" removed the window from the tiling_layer and added it to the floating, unmaximizing restored the previous state, so it re-added the window to the tiling layer and then the check breaks as the window is mapped twice (which should never happen).

I added a check if tiling-mode is still enabled to all cases, where we map stuff to the tiling_layer later again.
